src/NBomber/Api/CSharp.fs

```diff
@@ -209,6 +210,10 @@ type ScenarioBuilder =
     static member WithCustomStepExecControl(scenario: Scenario, execControl: Func<IStepExecControlContext voption, string voption>) =
         scenario |> FSharp.Scenario.withCustomStepExecControl(execControl.Invoke)
 
+    [<Extension>]
+    static member WithThresholds(scenario: Scenario, [<ParamArray>]thresholds: Metric[]) =
+        scenario |> FSharp.Scenario.withThresholds(Seq.toList thresholds)
```

Comment:
Will need to add a documentation comment to this and the other new methods in `Api`.